app/assets/javascripts/modules/namespaces/components/new-form.js

```diff
@@ -65,40 +68,26 @@ export default {
       name: {
         required,
         format(value) {
-          // extracted from models/namespace.rb
-          const regexp = /^[a-z0-9]+(?:[._-][a-z0-9]+)*$/;
-
-          // required already taking care of this
-          if (value === '') {
-            return true;
-          }
-
-          return regexp.test(value);
-        },
-        available(value) {
-          clearTimeout(this.timeout.name);
+          clearTimeout(this.timeout.validate);
 
           // required already taking care of this
           if (value === '') {
             return true;
           }
 
           return new Promise((resolve) => {
-            const searchName = () => {
-              const promise = NamespacesService.existsByName(value);
+            const validate = () => {
+              const promise = NamespacesService.validate(value);
 
-              promise.then((exists) => {
-                // leave it for the back-end
-                if (exists === null) {
-                  resolve(true);
+              promise.then((data) => {
+                if (!data.valid) {
```

Comment:
We need to clean `this.errors` from a previous validation if it's valid. If we don't do that errors messages will still be shown even if data is valid.

We can just simply

```js
set(this, 'errors', data.messages);
```

without any condition here. We already check for errors in the template, so we can do this check if it's empty there.